File: backend/src/main/java/com/pgvaale/backend/entity/PG.java
package com.pgvaale.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

/** PG ENTITY */
@Entity
@Table(name = "pgs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PG {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "pg_seq")
    @SequenceGenerator(name = "pg_seq", sequenceName = "pg_sequence", initialValue = 9000, allocationSize = 1)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private Owner owner;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User registeredUser;

    // ✅ List of image paths (max 5 should be handled via validation in controller
    // or frontend)
    @ElementCollection
    @CollectionTable(name = "pg_images", joinColumns = @JoinColumn(name = "pg_id"))
    @Column(name = "image_path")
    private List<String> imagePaths;

    private double latitude;
    private double longitude;

    private String amenities; // e.g., "AC,Gas,Fridge"
    private String nearbyResources; // e.g., "Hospital,Gym,Garden"

    private double rent;

    private String generalPreference;

    private String region; // Added region field
    private String availability;
}

File: PGVaaleDotNetBackend/Entities/PG.cs
using System.ComponentModel.DataAnnotations;

namespace PGVaaleDotNetBackend.Entities
{
    public class PG
    {
        [Key]
        public long Id { get; set; }

        [Required]
        public long OwnerId { get; set; }

        [Required]
        public long UserId { get; set; }

        // List of image paths (max 5 should be handled via validation in controller or frontend)
        public string ImagePaths { get; set; } = string.Empty; // Comma-separated list

        public double Latitude { get; set; }

        public double Longitude { get; set; }

        public string Amenities { get; set; } = string.Empty; // e.g., "AC,Gas,Fridge"

        public string NearbyResources { get; set; } = string.Empty; // e.g., "Hospital,Gym,Garden"

        public double Rent { get; set; }

        public string GeneralPreference { get; set; } = string.Empty;

        public string Region { get; set; } = string.Empty; // Added region field

        public string Availability { get; set; } = string.Empty;

        // Navigation properties
        public Owner Owner { get; set; } = null!;
        public User RegisteredUser { get; set; } = null!;
    }
}
